tools/custom-action/Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM ubuntu:23.04
+
+ENV LANG=en_US.UTF-8
+
+RUN \
+  export DEBIAN_FRONTEND=noninteractive && \
+  apt update -y && \
+  apt install -y locales && \
+  locale-gen en_US.UTF-8 && \
+  sysctl -w kernel.unprivileged_userns_clone=1 && \
+  apt install -y \
+    nodejs \
+    npm \
+    git \
+    sudo \
+    libgtk-3-dev \
+    libnotify-dev \
+    libgconf-2-4 \
+    libnss3 \
+    libxss1 \
+    libasound2 \
+    python3-full \
+    python3-pip \
+    rsync
+
+RUN \
+  export DEBIAN_FRONTEND=noninteractive && \
+  python3 -m pip install --break-system-packages \
+    bikeshed==3.13.1 \
+    tree_sitter==0.20.1 && \
+  npm install -g @mermaid-js/mermaid-cli@9.1.4 && \
+  npx --yes -- @mermaid-js/mermaid-cli@9.1.4 --version && \
```

Comment:
`npx --yes` implicitly installs the package, just without putting it in the PATH, so I don't think we need `npm install -g` as well unless we need to get the package into the PATH (we shouldn't).

Reply:
@kainino0x I didn't have good luck with it, after I did npx --yes, when the command was run in the action using container, it did network request again, but when I did global install, that have ceased to be an issue. Would it be possible to accommodate this redundancy in container environment to make sure those network requests are not sent during the CI run? I will replicate behavior in my branch and link the relevant run here.

Reply:
@kainino0x Please refer to following. If it is possible to get npx --yes to behave well, I am 100% for it, but unless, I think PATH in container is probably not very problematic as it is an isolated environment.

With global installation: https://github.com/mehmetoguzderin/gpuweb/actions/runs/5517031552/jobs/10059111737
Without global installation: https://github.com/mehmetoguzderin/gpuweb/actions/runs/5517146717/jobs/10059371153

It can't execute mermaid or others

Reply:
OK, I'm surprised by that, but agree installing globally isn't a problem.